pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/StandardIndexes.java

```diff
@@ -39,21 +39,64 @@
  *   included in a Pinot distribution. {@link StandardIndexes} contains one get method for each standard index,
  *   providing a typesafe way to get these references. Instead of having to write something like
  *   {@code (IndexType<BloomFilterConfig, BloomFilterReader, IndexCreator>)
- *   IndexService.getInstance().getOrThrow("bloom_filter")},
+ *   IndexService.getInstance().get("bloom_filter")},
  *   a caller can simply use {@link StandardIndexes#bloomFilter()}
  * </p>
  */
 @SuppressWarnings("unchecked")
 public class StandardIndexes {
+  public static final String FORWARD_ID = "forward_index";
```

Comment:
It was important because there were some code that iterate over the constants in a very specific order. In these cases I kept the order by explicitly introduce an order in the iteration, which is more verbose, but IMHO better as it explicitly introduce the order where the order really matters instead of implicitly use an order defined somewhere else.

But as you said, I had to deal with that in #10184. This is one of the reasons that PR is going to be trickier to merge than this one.